analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -49,19 +49,21 @@ public static SyntaxTree GetSyntaxTree<TLanguageKindEnum>(this CodeBlockStartAna
         public static SyntaxTree GetSyntaxTree(this SemanticModelAnalysisContext context) =>
             context.SemanticModel.SyntaxTree;
 
-
+        // ToDo: by default, do not verify if the project is Test or not when reporting. This is already being done at rule registration.
```

Comment:
Can you manually measure compile time and the impact of this check on some IT project? To measure if it is a problem or not. (Out of scope of this specific PR)

Reply:
I should as a next step

Reply:
I did some tests with `NodaTime`, which is running Scanner BEGIN and then `MSBuild.exe /t:"Restore,Rebuild" .\src\NodaTime.sln > out.txt`

This PR
>Time Elapsed 00:00:47.27
Time Elapsed 00:00:46.98
Time Elapsed 00:00:44.83

Master
>Time Elapsed 00:00:51.10
Time Elapsed 00:00:50.30
Time Elapsed 00:00:50.15

I guess I should check with other projects as well, but from an initial test, it seems that (read&cache during initialization + read during reporting) performs better than (iterating through assembly references all the time).